app/services/measure_unit_service.rb

```diff
@@ -0,0 +1,29 @@
+class MeasureUnitService
+  def initialize(measures)
+    @measures = measures
+  end
+
+  def call
+    units.each_with_object({}) do |unit, acc|
+      unit_key = "#{unit[:measurement_unit_code]}#{unit[:measurement_unit_qualifier_code]}"
+
+      if acc[unit_key].present?
+        acc[unit_key]['measure_sids'].add(unit[:measure_sid])
+
+        next
```

Comment:
Do you need next here?

Reply:
No, since we're at the tail of this iteration - good spot